### PR TITLE
Feature centralized logging

### PR DESCRIPTION
**Pull request by webatspeed**
Log settings mainly in
- logback-spring.xml
- CorrelationIdFilter.java
- application.properties
- pom.xml

Simplified log calling in
- LogService.java
- all other files

### CODE REVIEW

File: src/main/java/de/caritas/cob/userservice/filter/CorrelationIdFilter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import javax.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/userservice/filter/CorrelationIdFilter.java]: Couldn´t the @RequiredArgsConstructor be removed as there is no required arg in this class?

File: src/test/java/de/caritas/cob/userservice/api/facade/EmailNotificationFacadeTest.java [hunk 2]
```diff
@@ -228,7 +229,11 @@ public void setup() throws NoSuchFieldException, SecurityException {
         FIELD_VALUE_ROCKET_CHAT_SYSTEM_USER_ID);
     ReflectionTestUtils.setField(emailNotificationFacade, APPLICATION_BASE_URL_FIELD_NAME,
         APPLICATION_BASE_URL);
-    setInternalState(LogService.class, "LOGGER", logger);
+    //setInternalState(LogService.class, "LOGGER", logger);
```
Comment (maeventer) [inline src/test/java/de/caritas/cob/userservice/api/facade/EmailNotificationFacadeTest.java]: Please remove the line if it´s not needed any more

File: src/test/java/de/caritas/cob/userservice/api/facade/assignsession/AssignSessionFacadeTest.java [hunk 3]
```diff
@@ -77,20 +80,24 @@
...
 
-    assignSessionFacade.assignSession(FEEDBACKSESSION_WITH_CONSULTANT, CONSULTANT_WITH_AGENCY);
+    var thrown = assertThrows(InternalServerErrorException.class,
+        () -> assignSessionFacade.assignSession(FEEDBACKSESSION_WITH_CONSULTANT,
+            CONSULTANT_WITH_AGENCY)
+    );
 
-    verify(logService, times(1)).logInternalServerError(anyString(), any());
+    assertEquals(exception.getMessage(), thrown.getMessage());
+    //TODO: check internal server error log
```
Comment (maeventer) [inline src/test/java/de/caritas/cob/userservice/api/facade/assignsession/AssignSessionFacadeTest.java]: I´m not sure if the checks in the previous code have ever been executed. I think if the "expected" in the @Test annotation is used, the test is finished as soon as the Exception is thrown and no further code is executed. I think we should check if the verifications in your TODO´s are still needed.

Comment (webatspeed) [reply src/test/java/de/caritas/cob/userservice/api/facade/assignsession/AssignSessionFacadeTest.java]: Your assumption is right, no code had been reached after the call took place. I could not find a call of rollbackRemoveUsersFromRocketChatGroup and wasn't sure it's needed here. Do you know? The first todo is hard to check, currently. An easy solution would be just not to do it. :)

Comment (maeventer) [reply src/test/java/de/caritas/cob/userservice/api/facade/assignsession/AssignSessionFacadeTest.java]: Fine for me, so just remove these two verifications

**Review by maeventer — APPROVED** [review]

**Pull request merged**